packages/edit-site/src/components/block-editor/editor-canvas.js

```diff
@@ -31,33 +32,35 @@ function EditorCanvas( { enableResizing, settings, children, ...props } ) {
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
 	const deviceStyles = useResizeCanvas( deviceType );
 	const mouseMoveTypingRef = useMouseMoveTypingReset();
+
+	const editorStyles = useMemo( () => {
+		if ( settings.styles ) {
+			return [
+				...settings.styles,
+				{
+					css:
+						// Forming a "block formatting context" to prevent margin collapsing.
+						// @see https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Block_formatting_context
+						`.is-root-container{display:flow-root;${
+							// Some themes will have `min-height: 100vh` for the root container,
+							// which isn't a requirement in auto resize mode.
+							enableResizing ? 'min-height:0!important;' : ''
+						}}body{position:relative; ${
+							canvasMode === 'view' ? 'cursor: pointer;' : ''
+						}}}`,
+				},
+			];
+		}
+
+		return settings.styles;
+	}, [ settings.styles ] );
```

Comment:
I ended up doing a bit or refactoring here, so there's fewer `<style>` tags all over the place.